snapcraft/tests/test_lifecycle.py

```diff
@@ -659,12 +661,12 @@ def test_pull_is_dirty_if_target_arch_changes(
             self.fake_logger.output,
             Equals("Setting target machine to 'armhf'\n"))
 
+        self.assertThat(raised.step, Equals('pull'))
+        self.assertThat(raised.part, Equals('part1'))
         self.assertThat(
-            str(raised), Equals(
-                "The 'pull' step of 'part1' is out of date:\n"
-                "The 'deb_arch' project option appears to have changed.\n"
-                "In order to continue, please clean that part's 'pull' step "
-                "by running:\nsnapcraft clean part1 -s pull\n"))
+            raised.report,
```

Comment:
maybe do some line cleanup on these such that we get
```yaml
self.assertThat(raised.report, Equals(
     "The 'deb_arch' project option appears to have changed.\n"))
```

to make it easier on the eyes wrt indentation